### PR TITLE
Drop font size mult

### LINKED ISSUE #56: Upgrade to rebass 3
It seems like there are new versions of rebass and styled-system.

cc @sarathms

### LINKED ISSUE #57: Fix issue with fontSizeMult
We have a warning message saying that fontSizeMult is deprecated which we should fix.

I think this should be done with https://github.com/ooni/design-system/issues/56 and we should evaluate even just using the stock rebass Button component directly or drawing inspiration from it to implement a compatible API.

It has the `variant` prop which is equivalent to our `hollow` prop.

cc @sarathms

### LINKED ISSUE #95: Remove deprecated styled-components API innerRef in Radio component
A bunch of these warnings were seen in the OONI Explorer integration test logs.

> The "innerRef" API has been removed in styled-components v4 in favor of React 16 ref forwarding, use "ref" instead like a typical component. "innerRef" was detected on component "Radio__ErasedInputRadio".

### PR DESCRIPTION
**Pull request by hellais**
Fixes #56
Fixes #57 
Fixes #95

### CODE REVIEW

**Comment by sarathms** [comment]
This also includes porting all the components to use `rebass@v3`. Current progress:

- [ ] Text
- [ ] Button
- [ ] Input
- [ ] Container
- [ ] Heading
- [ ] IconButton
- [ ] Link
- [ ] Radio
- [ ] Select
- [ ] ShareButton
- [ ] Card
- [ ] InputWithIconButton
- [ ] Hero
- [ ] Modal

File: components/Button.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
+import PropTypes from 'prop-types'
```
Comment (hellais) [inline components/Button.js:1]: Yes this is a good idea to reduce the atom/molecule nesting hell I created!

File: components/atoms/ShareButton.js [hunk 2]
```diff
@@ -1,117 +1,62 @@
-import React from 'react'
```
Comment (hellais) [inline components/atoms/ShareButton.js:1]: Should this maybe also be moved out of the atoms directory tree?

File: components/atoms/Text.js [hunk 3]
```diff
@@ -1,30 +1,8 @@
...
-import {
-  space,
-  fontSize,
-  fontWeight,
-  textStyle,
-  textAlign,
-  color,
-  width
-} from '../util'
+const Text = props => (
```
Comment (hellais) [inline components/atoms/Text.js:4]: Why don't we just import this and do the override inside of the main component index?

**Comment by sarathms** [comment]
I just ran `yarn test` and the test for `Button` fails. We broke support for a `border` prop to render bordered buttons.

File: components/atoms/Text.js (hunk 3 above)
Comment (sarathms) [reply components/atoms/Text.js:4]: I was anticipating to come back and add some customization after running tests or review. So I left the file here. We can just directly export it like we do for `Flex` and `Box`. It is easier to maintain any overrides in its own file.

**Comment by hellais** [comment]
> I just ran yarn test and the test for Button fails. We broke support for a border prop to render bordered buttons.

Is there a way to unbreak this? If not do we have software that is using the bordered buttons?

I say we can also just merge this and avoid upgrading broken repos that make use of it, until we have a fix for it, by documenting it as a known bug.

**Comment by sarathms** [comment]
> Is there a way to unbreak this?

I actually already fixed it in b944cb4.  I just need to merge and release it.

**Pull request merged**